ccl/serverccl: TestTenantGRPCServices failed

ccl/serverccl.TestTenantGRPCServices [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3296858&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3296858&tab=artifacts#/) on master @ [3928f1963833fbf51ae47bd2a42ae6a200ebbb14](https://github.com/cockroachdb/cockroach/commits/3928f1963833fbf51ae47bd2a42ae6a200ebbb14):


```
=== RUN   TestTenantGRPCServices
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTenantGRPCServices582200065
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestTenantGRPCServices
    tenant_grpc_test.go:152: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTenantGRPCServices582200065
--- FAIL: TestTenantGRPCServices (20.23s)
=== RUN   TestTenantGRPCServices/statements_endpoint_fans_out_request_to_multiple_pods
    tenant_grpc_test.go:93: 
        	Error Trace:	tenant_grpc_test.go:93
        	Error:      	Received unexpected error:
        	            	Get "http://127.0.0.1:38797/_status/statements": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
        	Test:       	TestTenantGRPCServices/statements_endpoint_fans_out_request_to_multiple_pods
    --- FAIL: TestTenantGRPCServices/statements_endpoint_fans_out_request_to_multiple_pods (3.00s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestTenantGRPCServices PKG=./pkg/ccl/serverccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</details>
/cc @cockroachdb/server dhartunian
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantGRPCServices.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/serverccl.TestTenantGRPCServices [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3323574&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3323574&tab=artifacts#/) on master @ [4d6d79d8fcba33fc273297472d3c5e373b3c2b6c](https://github.com/cockroachdb/cockroach/commits/4d6d79d8fcba33fc273297472d3c5e373b3c2b6c):


```
=== RUN   TestTenantGRPCServices
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTenantGRPCServices242449592
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestTenantGRPCServices
    tenant_grpc_test.go:152: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTenantGRPCServices242449592
--- FAIL: TestTenantGRPCServices (5.63s)
=== RUN   TestTenantGRPCServices/statements_endpoint_fans_out_request_to_multiple_pods
    tenant_grpc_test.go:93: 
        	Error Trace:	tenant_grpc_test.go:93
        	Error:      	Received unexpected error:
        	            	Get "http://127.0.0.1:37127/_status/statements": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
        	Test:       	TestTenantGRPCServices/statements_endpoint_fans_out_request_to_multiple_pods
    --- FAIL: TestTenantGRPCServices/statements_endpoint_fans_out_request_to_multiple_pods (3.00s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestTenantGRPCServices PKG=./pkg/ccl/serverccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/server dhartunian
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantGRPCServices.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/serverccl/tenant_grpc_test.go
+++ b/pkg/ccl/serverccl/tenant_grpc_test.go
@@ -13,6 +13,7 @@ import (
 	"io/ioutil"
 	"net/http"
 	"testing"
+	"time"
 
 	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
@@ -38,6 +39,7 @@ func TestTenantGRPCServices(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	ctx := context.Background()
+	httpClient := httputil.NewClientWithTimeout(15 * time.Second)
 
 	serverParams, _ := tests.CreateTestServerParams()
 	testCluster := serverutils.StartNewTestCluster(t, 3, base.TestClusterArgs{
@@ -77,7 +79,7 @@ func TestTenantGRPCServices(t *testing.T) {
 	})
 
 	t.Run("gRPC Gateway is running", func(t *testing.T) {
-		resp, err := httputil.Get(ctx, "http://"+tenant.HTTPAddr()+"/_status/statements")
+		resp, err := httpClient.Get(ctx, "http://"+tenant.HTTPAddr()+"/_status/statements")
 		defer http.DefaultClient.CloseIdleConnections()
 		require.NoError(t, err)
 		defer resp.Body.Close()
@@ -98,7 +100,7 @@ func TestTenantGRPCServices(t *testing.T) {
 	defer connTenant2.Close()
 
 	t.Run("statements endpoint fans out request to multiple pods", func(t *testing.T) {
-		resp, err := httputil.Get(ctx, "http://"+tenant2.HTTPAddr()+"/_status/statements")
+		resp, err := httpClient.Get(ctx, "http://"+tenant2.HTTPAddr()+"/_status/statements")
 		defer http.DefaultClient.CloseIdleConnections()
 		require.NoError(t, err)
 		defer resp.Body.Close()
@@ -115,7 +117,7 @@ func TestTenantGRPCServices(t *testing.T) {
 	defer connTenant3.Close()
 
 	t.Run("fanout of statements endpoint is segregated by tenant", func(t *testing.T) {
-		resp, err := httputil.Get(ctx, "http://"+tenant3.HTTPAddr()+"/_status/statements")
+		resp, err := httpClient.Get(ctx, "http://"+tenant3.HTTPAddr()+"/_status/statements")
 		defer http.DefaultClient.CloseIdleConnections()
 		require.NoError(t, err)
 		defer resp.Body.Close()
@@ -160,7 +162,7 @@ func TestTenantGRPCServices(t *testing.T) {
 	})
 
 	t.Run("sessions endpoint is available", func(t *testing.T) {
-		resp, err := httputil.Get(ctx, "http://"+tenant.HTTPAddr()+"/_status/sessions")
+		resp, err := httpClient.Get(ctx, "http://"+tenant.HTTPAddr()+"/_status/sessions")
 		defer http.DefaultClient.CloseIdleConnections()
 		require.NoError(t, err)
 		require.Equal(t, 200, resp.StatusCode)

